src/mame/jaleco/megasys1.cpp

```diff
@@ -365,6 +389,38 @@ void megasys1_state::megasys1B_monkelf_map(address_map &map)
 #define INTERRUPT_NUM_C INTERRUPT_NUM_B
 #define interrupt_C     interrupt_B
 
+void megasys1_state::megasys1c_handle_scanline_irq(int scanline)
+{
+	if(scanline == 224) // vblank-out irq
+		m_maincpu->set_input_line(4, HOLD_LINE);
+
+	if(scanline == 80)
+		m_maincpu->set_input_line(1, HOLD_LINE);
+}
+
+TIMER_DEVICE_CALLBACK_MEMBER(megasys1_state::megasys1C_scanline)
+{
+	int scanline = param;
+
+	megasys1c_handle_scanline_irq(scanline);
+
+	if(scanline == 0)
+		m_maincpu->set_input_line(2, HOLD_LINE);
+
+}
+TIMER_DEVICE_CALLBACK_MEMBER(megasys1_bc_iomcu_state::megasys1C_bigstrik_scanline)
+{
+	int scanline = param;
+
+	megasys1c_handle_scanline_irq(scanline);
+
+	if(scanline == 0) // end of vblank (rising edge)
+	{
+		LOG("%s: megasys1C_bigstrik_scanline: Send INT1 to MCU: (scanline %03d)\n", machine().describe_context(), scanline);
+		m_iomcu->set_input_line(INPUT_LINE_IRQ1, ASSERT_LINE);
+	}
```

Comment:
When does this get deasserted?  I can’t see it at a glance.  Is this the reason for the hack in the MCU?

Reply:
We can change it to `HOLD_LINE`.

Reply:
But how does it actually get deasserted in real life on the hardware?  Does it detect an interrupt acknowledge cycle (which is what `HOLD_LINE` is for), or does it generate a fixed length pulse (which can be done `pulse_input_line`), or does something else cause the signal to be deasserted?

Reply:
In this case, the IRQ1 signal on the TLCS90 is, in fact, tied to VBlank signal on actual pcb. The interrupt is requested when VBlank ends (scanline == 0) and the line is desasserted when VBlank starts again (scanline == 224).
So, according of what you state above, we may need to simply put a CLEAR_LINE on the timer callback when scanline == 224

Reply:
> So, according of what you state above, we may need to simply put a CLEAR_LINE on the timer callback when scanline == 224

Correct.  That’s the proper way to implement it in this case.